README.md

```diff
@@ -1,5 +1,122 @@
+* **External Care Provider**: another care organization participating in Shared Care Planning.
+  * **Care Plan Service** is used by *Care Plan Contributor*s to create CarePlans, and manages CareTeams according to those CarePlans. 
+  * **Care Plan Contributor** provides access to its FHIR API to the local ORCA instance, and access the local FHIR API through ORCA.
+  * **Authorization Server** (see above)
+
+### Transactions
+This section describes how the Shared Care Planning transactions are implemented in ORCA.
+
+#### Creating a new CarePlan
+1. The care professional using the organization's EHR, selects a patient and launches the ORCA *Frontend* web application through *Orchestrator*.
+2. *Frontend* retrieves and shows the following data through ORCA's *Orchestrator*:
+   - the launch context: the FHIR Patient, ServiceRequest, and PractitionerRole resources.
+     This is retrieved through *Orchestrator* from the local EHR's FHIR API.
+   - the existing CarePlans for the patient.
+     This is retrieved through *Orchestrator* from the *Care Plan Service*, which is either local or remote.
+3. The care professional inputs the CarePlan details (name, condition) and submits.
+4. The *Frontend* creates the new CarePlan resource at the *Care Plan Service* through ORCA *Orchestrator*. 
+
+#### Workflow initiation/acceptance
+When a care professional wants to initiate a FHIR Task for another care organization, they start a new FHIR workflow.
+
+The process is as follows:
+1. Placer: the care professional, using the *Frontend*, chooses to create a new Task for a specified Condition.
+2. Placer: *Frontend* creates a new FHIR Task at the Care Plan Service through ORCA's *Orchestrator*.
+3. At this point, the Care Plan Service notifies the Task filler that a new Task is available.
+4. The Task filler and placer now negotiate the Task details:
+   1. Filler: if the filler needs more information, it adds a Questionnaire to the `Task.input`.
+   2. Placer: responds by filling the Questionnare, adding a QuestionnaireResponse to the Task.output.
+   3. Filler: verifies the QuestionnaireResponse either:
+      - Accept the Task if it is able and willing to perform it.
+      - Add another Questionnaire to the `Task.input` if it needs more information.
+      - Reject the Task if it cannot/won't perform it.
+
+Whenever a Task is created/updated at the Care Plan Service, it notifies the other Task participant.
+The notification is then handled to perform the Task negotiation:
+- Filler: notification is received by ORCA's *Orchestrator* and forwarded to the *Task Engine* (TODO).
+  The *Task Engine* then decides whether to accept the Task, and if not, what additional information is required.
+- Placer: notification is received by ORCA's *Orchestrator* and forwarded to the *Task Engine* (TODO).
+  The *Frontend* has a websocket connection to the *Task Engine* to receive updates on the Task status.
```

Comment:
I think it's probably easier to use something like an `EventSource`  to stream data from the server to the client, we only need to receive data, and it's much easier to implement. 

```suggestion
  The *Frontend* is notified by the *Task Engine* when the Task status changes.
```